### PR TITLE
Test sts

### PR DESCRIPTION
**Pull request by jderusse**
Add tests on STS

### CODE REVIEW

File: src/CodeGenerator/Generator/RequestSerializer/QuerySerializer.php [hunk 1]
```diff
@@ -123,15 +150,13 @@ private function dumpArrayStructure(string $output, string $input, StructureShap
         $memberCode = implode("\n", array_map(function (StructureMember $member) use ($output) {
             $shape = $member->getShape();
             if ($member->isRequired() || $shape instanceof ListShape || $shape instanceof MapShape) {
-                $inputElement = '$input->get' . $member->getName() . '()';
                 $body = 'MEMBER_CODE';
+                $inputElement = '$input->get' . $member->getName() . '()';
             } else {
+                $body = 'if (null !== $v = INPUT_NAME) {
```
Comment (jderusse) [inline src/CodeGenerator/Generator/RequestSerializer/QuerySerializer.php]: not related to that PR, but consistent with `rest-json`

File: src/Core/Sts/Input/AssumeRoleRequest.php [hunk 2]
```diff
@@ -198,58 +198,53 @@ public function requestBody(): string
...
+            $indices->kfc822c1 = 0;
             foreach ($input as $value) {
-                ++$indices->kd8fbed2;
+                ++$indices->kfc822c1;
 
                 if (null !== $value) {
                     (static function (PolicyDescriptorType $input) use (&$payload, $indices) {
                         if (null !== $v = $input->getarn()) {
-                            $payload["PolicyArns.{$indices->kd8fbed2}.arn"] = $v;
+                            $payload["PolicyArns.member.{$indices->kfc822c1}.arn"] = $v;
```
Comment (jderusse) [inline src/Core/Sts/Input/AssumeRoleRequest.php]: this is the bug spotted in Sts tests

File: src/Core/Tests/Unit/Input/AssumeRoleRequestTest.php [hunk 3]
```diff
@@ -11,31 +11,49 @@ class AssumeRoleRequestTest extends TestCase
...
             ])],
-            'TransitiveTagKeys' => ['change me'],
-            'ExternalId' => 'change me',
-            'SerialNumber' => 'change me',
-            'TokenCode' => 'change me',
+            'TransitiveTagKeys' => ['Project', 'Cost-Center'],
+            'ExternalId' => '123ABC',
         ]);
 
+        /** @see https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole.html */
```
Comment (jderusse) [inline src/Core/Tests/Unit/Input/AssumeRoleRequestTest.php]: I think such comment is helpfull to understand where the "expectation" come from

File: src/Lambda/Input/PublishLayerVersionRequest.php [hunk 4]
```diff
@@ -125,7 +125,7 @@ public function requestBody(): string
 
         (static function (array $input) use (&$payload) {
             foreach ($input as $value) {
-                $payload['CompatibleRuntimes'][] = $value;
+                $payload['CompatibleRuntimes']['member'][] = $value;
```
Comment (jderusse) [inline src/Lambda/Input/PublishLayerVersionRequest.php]: This has been changed by my fix. Can you confirm?

File: src/Ses/Tests/Unit/SesClientTest.php [hunk 5]
```diff
@@ -28,7 +28,7 @@ public function testSendMessage()
             $content = json_decode($options['body'], true);
 
             $this->assertSame('Hello!', $content['Content']['Simple']['Subject']['Data']);
-            $this->assertSame('tobias.nyholm@gmail.com', $content['Destination']['ToAddresses'][0]);
+            $this->assertSame('tobias.nyholm@gmail.com', $content['Destination']['ToAddresses']['member'][0]);
```
Comment (jderusse) [inline src/Ses/Tests/Unit/SesClientTest.php]: indeed, `ToAddresses` contains a `member` key => https://docs.aws.amazon.com/ses/latest/APIReference/API_Destination.html

File: src/Sns/Input/PublishInput.php [hunk 6]
```diff
@@ -134,41 +134,34 @@ public function requestBody(): string
...
         }
 
         (static function (array $input) use (&$payload, $indices) {
-            $indices->kb0b4646 = 0;
+            $indices->kd0379ae = 0;
             foreach ($input as $key => $value) {
-                ++$indices->kb0b4646;
-                $payload["MessageAttributes.{$indices->kb0b4646}.Name"] = $key;
+                ++$indices->kd0379ae;
+                $payload["MessageAttributes.entry.{$indices->kd0379ae}.Name"] = $key;
```
Comment (jderusse) [inline src/Sns/Input/PublishInput.php]: List contains a `member` key, and map contains a `entry` key (see https://docs.aws.amazon.com/fr_fr/sns/latest/api/API_Publish.html)

File: src/CodeGenerator/Generator/RequestSerializer/QuerySerializer.php (hunk 1 above)
Comment (Nyholm) [reply src/CodeGenerator/Generator/RequestSerializer/QuerySerializer.php]: Thank you

File: src/Core/Sts/Input/AssumeRoleRequest.php (hunk 2 above)
Comment (Nyholm) [reply src/Core/Sts/Input/AssumeRoleRequest.php]: Awesome. That is the one you were mentioning before but I didnt understand =)

File: src/Core/Tests/Unit/Input/AssumeRoleRequestTest.php (hunk 3 above)
Comment (Nyholm) [reply src/Core/Tests/Unit/Input/AssumeRoleRequestTest.php]: 👍 

I added data before but that was from request I made myself.

File: src/Lambda/Input/PublishLayerVersionRequest.php (hunk 4 above)
Comment (Nyholm) [reply src/Lambda/Input/PublishLayerVersionRequest.php]: Yes. It looks correct. 

I checked with the specification.

File: src/Ses/Tests/Unit/SesClientTest.php (hunk 5 above)
Comment (Nyholm) [reply src/Ses/Tests/Unit/SesClientTest.php]: Good. Correct!

File: src/Sns/Input/PublishInput.php (hunk 6 above)
Comment (Nyholm) [reply src/Sns/Input/PublishInput.php]: Great catch!

File: src/Sqs/Input/SendMessageRequest.php [hunk 7]
```diff
@@ -193,7 +191,6 @@ public function requestBody(): string
                         if (null !== $v = $input->getStringValue()) {
                             $payload["MessageSystemAttribute.{$indices->k6857220}.Value.StringValue"] = $v;
                         }
-
```
Comment (Nyholm) [inline src/Sqs/Input/SendMessageRequest.php]: I added some space here before. It made it a little easier to read..

What we should do is to add a comment here os something. Like 

```
// $input->getBinartValue()
```

That will help with readability. But that is another PR.

**Pull request merged**